### PR TITLE
[Sprint 7] Implement api delete and edit request leave

### PR DESCRIPTION
**Pull request by HyperionsThs**
[Sprint 7] Implement api delete and edit request leave
Link : https://dev.framgia.com/redmine/issues/126580
Effect : N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java [hunk 1]
```diff
@@ -142,6 +149,16 @@ private void initRequest(LeaveRequest requestLeave) {
         }
     }
 
+    private int searchLeaveTypePosition(List<com.framgia.wsm.data.model.LeaveType> leaveTypesList,
```
Comment (tranminhtri911) [inline app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java]: cai gi day :v

Comment (HyperionsThs) [inline app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java]: lấy id để hiển thị lại trên cái dialog @@

File: app/src/main/java/com/framgia/wsm/data/source/RequestDataSource.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
 
         Observable<Object> deleteFormRequestLeave(@NonNull int requestLeaveId);
 
-        Observable<BaseResponse<LeaveRequest>> editFormRequestLeave(
-                @NonNull LeaveRequest requestLeave);
+        Observable<RequestLeaveRequest> editFormRequestLeave(@NonNull int requestId,
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/source/RequestDataSource.java]: ? 
pls send to me json when editFromRequestLeave success

File: app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java [hunk 3]
```diff
@@ -118,10 +121,14 @@
         mPresenter.getUser();
         initRequest(requestLeave);
         mActionType = actionType;
+        setVisibleGroup(true);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java]: .
setVisibleGroup(mActionType == ActionType.ACTION_CREATE)

**Pull request merged**